esp/services/ws_ecl/ws_ecl_service.cpp

```diff
@@ -274,19 +280,34 @@ void initBareMetalRoxieTargets(MapStringToMyClass<ISmartSocketFactory> &connMap,
         }
         else
         {
-            VStringBuffer xpath("Software/RoxieCluster[@name='%s']", process.str());
-            Owned<IPropertyTreeIterator> it = pRoot->getElements(xpath.str());
-            ForEach(*it)
+            const char *farmerPort = nullptr;
+            VStringBuffer xpath("Software/RoxieCluster[@name='%s'][1]", process.str());
```

Comment:
Looks like you change the logic here. The old code loops through every RoxieClutser settings with the same name. Now, if there are >1 RoxieClutser settings with the same name, only first one is read. I guess that there should not be >1 RoxieClutser settings with the same name. Please confirm.

Reply:
Yes, there should only be one with a given way.  If misconfigured thought picking the first one vs merging multiple was a better approach.